### PR TITLE
Enforce eager mode

### PR DESCRIPTION
**Pull request by danielholanda**
Closes https://github.com/groq/mlagility/issues/282

## Issue

You can find a description of the issue [here](https://github.com/groq/mlagility/issues/282).

## How this PR addresses the issue

This PR ensures that benchit only tries to process Pytorch models if they have **not** been compiled using `torch.compile()`.

The following message will be shown when a compiled Pytorch model in encountered:

```
compiled.py:
        model (executed 1x)
                Class:          LinearTestModel (<class 'compiled.LinearTestModel'>)
                Location:       /home/dhnoronha/.local/lib/python3.8/site-packages/torch/__init__.py, line 1392
                Parameters:     220 (<0.1 MB)
                Hash:           e33f2d21
                Status:         Skipping model compiled using torch.compile(). Benchit requires models to be in eager mode.
```

### CODE REVIEW

File: src/mlagility/analysis/analysis.py [hunk 1]
```diff
@@ -80,7 +80,6 @@ def call_benchit(
     """
     Calls the benchit function from within the model forward function
     """
-
```
Comment (jeremyfowers) [inline src/mlagility/analysis/analysis.py:83]: nit: any reason for removing this whitespace?

File: src/mlagility/analysis/analysis.py [hunk 2]
```diff
@@ -119,30 +121,34 @@ def call_benchit(
...
-            groqview=tracer_args.groqview,
-            sequence=tracer_args.sequence,
-            onnx_opset=tracer_args.onnx_opset,
-        )
-
-        if Action.BENCHMARK in tracer_args.actions:
-            model_info.status_message = "Model successfully benchmarked!"
-            model_info.performance = perf
+        if model_info.model_type == build.ModelType.PYTORCH_COMPILED:
+            model_info.status_message = "Skipping model compiled using torch.compile(). Benchit requires models to be in eager mode."
```
Comment (jeremyfowers) [inline src/mlagility/analysis/analysis.py:125]: What if `runtime` is set to `torch-compiled`? Is the idea that `torch-compiled` only works with eager mode models that we compile ourselves?

File: src/mlagility/analysis/analysis.py [hunk 3]
```diff
@@ -305,7 +314,10 @@ def explore_frame(
         if issubclass(type(local_var), torch.nn.Module):
             if type(local_var) in tracer_args.torch_activations:
                 return
-            model_type = build.ModelType.PYTORCH
+            if "dynamo_ctx" in local_var.__dict__:
```
Comment (jeremyfowers) [inline src/mlagility/analysis/analysis.py:317]: How durable is this? Is there a torch API for checking if a model is compiled?

Comment (jeremyfowers) [reply src/mlagility/analysis/analysis.py:317]: I couldn't find one... oh well, as long as your test provides coverage of this then we're good.

File: src/mlagility/analysis/analysis.py (hunk 2 above)
Comment (jeremyfowers) [reply src/mlagility/analysis/analysis.py:125]:
```suggestion
            model_info.status_message = "Skipping model compiled using torch.compile(). benchit requires models to be in eager mode (regardless of what runtime you have selected)."
```

little "b" for benchit

**Review by jeremyfowers — APPROVED** [review]

File: src/mlagility/analysis/analysis.py (hunk 2 above)
Comment (danielholanda) [reply src/mlagility/analysis/analysis.py]: Changed!

File: src/mlagility/analysis/analysis.py (hunk 1 above)
Comment (danielholanda) [reply src/mlagility/analysis/analysis.py]: Adding it back.

File: src/mlagility/analysis/analysis.py (hunk 3 above)
Comment (danielholanda) [reply src/mlagility/analysis/analysis.py:322]: Sounds good

**Pull request merged**